src/qa4sm_reader/plot_utils.py

```diff
@@ -33,11 +34,32 @@ def _get_grid(a):
     len_a = int((a_max - a_min) / da + 1)
     return a_min, a_max, da, len_a
 
+def _get_grid_for_irregulars(a, grid_stepsize):
+    "Find the stepsize of the grid behind a for datasets with predeifned grid stepsize, and return the parameters for that grid axis."
+    a = np.unique(a)
+    a_min = a[0]
+    a_max = a[-1]
+    da = grid_stepsize
+    len_a = int((a_max - a_min) / da + 1)
+    return a_min, a_max, da, len_a
+
 def _value2index(a, a_min, da):
     "Return the indexes corresponding to a. a and the returned index is a numpy array."
     return ((a - a_min) / da).astype('int')
 
-def geotraj_to_geo2d(df, var, index=globals.index_names):
+def oversample(lon, lat, data, extent, dx, dy):
+
+    other = BasicGrid(lon, lat)
+    reg_grid = genreg_grid(dx, dy, minlat=extent[2], maxlat=extent[3],
+                           minlon=extent[0], maxlon=extent[1])
+
+    lut = reg_grid.calc_lut(other, max_dist=25000)
```

Comment:
increasing this value, or removing the max_dist keyword (which would restore the pygeogrids<=0.3.0 behavior) should work

Reply:
then we can leave pygeogrids unpinned